bravado/http_future.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from bravado.exception import HTTPError
+
+DEFAULT_TIMEOUT_S = 5.0
+
+
+class HttpFuture(object):
+    """A future which inputs HTTP params"""
+
+    def __init__(self, future, response_adapter, callback):
+        """Kicks API call for Fido client
+
+        :param future: future object
+        :type future: :class: `concurrent.futures.Future`
+        :param response_adapter: Adapter which exposes json(), status_code()
+        :type response_adapter: :class: `bravado_core.response.ResponseLike`
+        :param callback: Function to be called on the response
+        """
+        self.future = future
+        self.response_adapter = response_adapter
+        self.response_callback = callback
+
+    def result(self, timeout=DEFAULT_TIMEOUT_S):
+        """Blocking call to wait for API response
+
+        :param timeout: Timeout in seconds for which client will get blocked
+        to receive the response
+        :return: swagger response return value when given a callback or the
+            http_response otherwise.
+        """
+        http_response = self.response_adapter(
+            self.future.result(timeout=timeout))
+
+        if self.response_callback:
+            swagger_return_value = self.response_callback(http_response)
+            return raise_http_error_based_on_status(
```

Comment:
from the method name, it looks like it should work like requests's `raise_for_status` i.e. return None if sucess else raise error. How about returning the `swagger_return_value` after calling `raise_http_error_based_on_status`?